### PR TITLE
Added pagination

### PR DESCRIPTION
**Pull request by Dmitry925**
Added pagination base classes and extention methods

### CODE REVIEW

**Review by svadrif — APPROVED** [review]

File: src/Application/Extentions/QueryableExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using Microsoft.EntityFrameworkCore;
+
+namespace Application.Extentions;
+
+public static class QueryableExtentions
+{
+    public static async Task<PagedList<TEntity>> ToPagedListAsync<TEntity>(this IQueryable<TEntity> source, PagedQueryBase query)
+        where TEntity : class
+    {
+        var count = source.Count();
```
Comment (svadrif) [inline src/Application/Extentions/QueryableExtensions.cs:11]: Can we use CountAsync here?

File: src/Application/Interfaces/IAppUserRepository.cs [hunk 2]
```diff
@@ -1,14 +1,11 @@
...
-using System.Threading.Tasks;
+﻿using Application.Pagination;
+using Domain.Entities;
 
 namespace Application.Interfaces
 {
     public interface IAppUserRepository : IGenericRepository<AppUser>
     {
         Task<AppUser> GetByTelegramId(long telegramId, bool tracking);
+        Task<PagedList<AppUser>> GetPagedAsync(PagedQueryBase query, bool tracking);
```
Comment (svadrif) [inline src/Application/Interfaces/IAppUserRepository.cs:9]: I think you should create another generic interface with GetPagedAsync generic method

Comment (Dmitry925) [reply src/Application/Interfaces/IAppUserRepository.cs:9]: like this?
image

File: src/Application/Extentions/QueryableExtensions.cs (hunk 1 above)
Comment (Dmitry925) [reply src/Application/Extentions/QueryableExtensions.cs]: Updated

**Review by svadrif — APPROVED** [review]

**Pull request merged**